action_server/src/robocorp/action_server/_server_expose.py

```diff
@@ -124,180 +180,92 @@ async def expose_server(
     datadir: str,
     api_key: str | None = None,
     expose_session: str | None = None,
+    ping_interval: int = 4,
 ):
     """
     Exposes the server to the world.
     """
 
     pong_queue: asyncio.Queue = asyncio.Queue(maxsize=1)
-    no_connection_queue: asyncio.Queue = asyncio.Queue(maxsize=1)
 
     async def listen_for_requests() -> None:
-        max_retries = 3
         retry_delay = 3
         retries = 0
 
         session_payload: Optional[SessionPayload] = (
             get_expose_session_payload(expose_session) if expose_session else None
         )
-        while retries < max_retries:
-            headers = (
-                {
-                    "x-session-id": session_payload.sessionId,
-                    "x-session-secret": session_payload.sessionSecret,
-                }
-                if session_payload
-                else {}
-            )
 
-            try:
-                async with websockets.connect(
-                    f"wss://client.{expose_url}",
-                    extra_headers=headers,
-                    logger=log,
-                    close_timeout=0,
-                ) as ws:
-                    if retries > 0:
-                        log.info("Reconnected to expose tunnel server")
-                        retries = 0
-
-                    ping_task = asyncio.create_task(
-                        handle_ping_pong(ws, pong_queue, no_connection_queue)
-                    )
-
-                    try:
-                        while True:
-                            done, pending = await asyncio.wait(
-                                [ws.recv(), no_connection_queue.get()],
-                                return_when=asyncio.FIRST_COMPLETED,
-                            )
-
-                            for task in done:
-                                message = task.result()
-                                if message == "no_connection":
-                                    raise ExposePingTimeoutError("No connection")
-
-                            for task in pending:
-                                task.cancel()
-
-                            if message == "pong":
-                                await pong_queue.put("pong")
-                                continue
+        headers = (
+            {
+                "x-session-id": session_payload.sessionId,
+                "x-session-secret": session_payload.sessionSecret,
+            }
+            if session_payload
+            else {}
+        )
+
+        async for ws in websockets.connect(
```

Comment:
Uses websockets [own mechanisms](https://websockets.readthedocs.io/en/stable/faq/client.html#how-do-i-reconnect-when-the-connection-drops) to reconnect.